### DISCUSSION TITLE
Issues with useTreeData hook

### DISCUSSION THREAD

**Discussion by IshakuMN** [adobe/react-spectrum#5768] [Q&A] [answered]
when I try to append or perform any adding function to a node in my tree, It adds double of the intended items  to add.  I see it through the console logs using tree.getItems(key). And under the children of a node I can only see the first added item even though I try to add multiple times it shows be only the first one I added. what could be the issue?  no issues in adding to the root of the tree only to a node

`let tree = useTreeData({
    initialItems: [
      {
        name: "People",
        items: [{ name: "David" }, { name: "Sam" }, { name: "Jane" }],
      },
      {
        name: "Animals",
        items: [{ name: "Aardvark" }, { name: "Kangaroo" }, { name: "Snake" }],
      },
    ],
    initialSelectedKeys: ["Sam", "Kangaroo"],
    getKey: (item) => item.name,
    getChildren: (item) => item.items,
  });

  const handleAddToTree = () => {
    tree.append("People", { name: "uuu" });
    console.log(tree.items, tree.getItem("People"));
  };`
  
  this is the result of calling the add function twice
  
<img width="604" alt="Screenshot 2024-01-30 at 00 36 21" src="https://github.com/adobe/react-spectrum/assets/104638451/2bf738db-aa8a-40b8-973d-9ea3f7cb3975">

**Comment by IshakuMN** [comment]
I see that I only have that issue in development and not in production

**Reply by reidbarber** [reply]
Is this caused by React [StrictMode](https://react.dev/reference/react/StrictMode)? Is your function being run multiple times? If this is inside a useEffect, it will run twice in strict mode.

**Reply by IshakuMN** [reply]
No, is not in React StrictMode and is not inside useEffect nor is it running multiple times. I seriously don't know why. I even created another protect but is still misbehaving and the misbehaviours are not even consistent. I tried so many things but only works on build.

**Reply by reidbarber** [reply]
If you're able to reproduce it in a codesandbox or stackblitz, I can take a closer look.

**Reply by IshakuMN** [reply]
Okay, here is the link. You can check what is going on in the console

https://codesandbox.io/p/devbox/autumn-architecture-xkcfjz?file=%2Ftsconfig.json&workspaceId=4313c52e-5a37-4dd6-9706-f94f179d4229

**Reply by reidbarber** [reply, marked as answer]
I added `reactStrictMode: false` to the `next.config.js` and the problem disappeared, but it should be working in strict mode (our test suite runs in strict mode.) Feel free to convert this into an issue so it can be further investigated.
